controlplane/src/core/repositories/SubgraphRepository.ts

```diff
@@ -679,4 +694,38 @@ export class SubgraphRepository {
 
     return subgraphDTOs;
   }
+
+  // returns the latest valid schema version of a subgraph
+  public async getLatestValidSchemaVersion(subgraphName: string) {
+    const latestValidVersion = await this.db
```

Comment:
We need to work with the fed graph name too

Reply:
Done